packages/search-input/src/SearchInput/SearchInput.tsx

```diff
@@ -0,0 +1,436 @@
+            }
+          };
+
+          return React.cloneElement(child, {
+            ...child.props,
+            id: `result-${index}`,
+            key: `result-${index}`,
+            ref: child.props.ref ?? resultRefs?.(`${index}`),
+            highlighted: index === highlightIndex,
+            onClick: onElementClick,
+          } as SearchResultProps);
+        } else if (isComponentType(child, 'SearchResultGroup')) {
+          const nestedChildren = React.Children.map(
+            child.props.children,
+            processChild,
+          );
+
+          if (nestedChildren && nestedChildren.length > 0) {
+            return React.cloneElement(child, {
+              ...child.props,
+              children: nestedChildren,
+            } as SearchResultGroupProps);
+          }
+        }
+      };
+
+      const updatedChildren = React.Children.map(children, processChild);
+
+      return {
+        resultsCount,
+        updatedChildren,
+      };
+    }, [children, highlightIndex, inputRef, resultRefs, changeInputValue]);
+
+    const { updatedChildren, resultsCount } = useMemo(
+      () => processChildren(),
+      [processChildren],
+    );
+
+    type Direction = 'next' | 'prev' | 'first' | 'last';
+    const updateHighlight = (direction: Direction) => {
+      switch (direction) {
+        case 'first': {
+          setHighlightIndex(0);
+          break;
+        }
+
+        case 'last': {
+          setHighlightIndex(resultsCount);
+          break;
+        }
+
+        case 'next': {
+          const nextIndex =
+            !isUndefined(highlightIndex) && highlightIndex + 1 < resultsCount
+              ? highlightIndex + 1
+              : 0;
+          setHighlightIndex(nextIndex);
+          break;
+        }
+
+        case 'prev': {
+          const nextIndex =
+            !isUndefined(highlightIndex) && highlightIndex - 1 >= 0
+              ? highlightIndex - 1
+              : resultsCount - 1;
+          setHighlightIndex(nextIndex);
+        }
+      }
+    };
+
+    /** Event Handlers */
+
+    const handleInputChange: ChangeEventHandler<HTMLInputElement> = e => {
+      handleChange?.(e);
+    };
+
+    const handleOpenMenuAction: EventHandler<SyntheticEvent<any>> = e => {
+      if (disabled) {
+        e.preventDefault();
+        e.stopPropagation();
+      } else {
+        openMenu();
+      }
+    };
+
+    const handleSearchBoxMousedown: MouseEventHandler<HTMLDivElement> = e => {
+      if (disabled) {
+        // Prevent container from gaining focus by default
+        e.preventDefault();
+      }
+    };
+
+    const handleSearchBoxClick: MouseEventHandler<HTMLDivElement> = e => {
+      if (e.target !== clearButtonRef.current) {
+        // Don't open the menu if it was the clear button that was clicked.
+        // Could use `e.stopPropagation` in clear button handler instead,
+        // but users may still want to handle that click event
+        handleOpenMenuAction(e);
+      }
+    };
```

Comment:
@brookescarlett Here I'm just checking the event target instead of stopping propagation